pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -530,12 +530,14 @@ func isNoOp(s state) bool {
 	return ok
 }
 
+// nolint: unparam
 func threadID(didCommMsg service.DIDCommMsg) (string, error) {
-	if didCommMsg.Type() == InvitationMsgType || didCommMsg.Type() == oobMsgType {
+	tID, err := didCommMsg.ThreadID()
```

Comment:
when is `didCommMsg.ThreadID()` returning an error? Why can't we remove `threadID()`?

Reply:
we can remove it, it returns an error when id and thread id is absent.

Reply:
Let's please remove it.